### ISSUE TITLE
V0.5.0 Test Plan 

### ISSUE THREAD

**Issue by yukirora** [microsoft/superbenchmark#340]
# V0.5.0 Test Plan

## Test Table

| Machine Type | #Node * #GPU * GPU Type | PyTorch Version | Accelerated Computing Toolkit | Status |
|---|---|---|---|---|
| ND A100 v4 | 1 * 8 * A100 40GB SXM | PyTorch 1.8 | CUDA 11.1 |  |
| NDm A100 v4 | 1 * 8 * A100 80GB SXM | PyTorch 1.8 | CUDA 11.1 |  |
| NDm A100 v4 | 2 * 8 * A100 80GB SXM | PyTorch 1.8 | CUDA 11.1 |  |
| Hayabusa  | 1* 16 * MI200 | PyTorch 1.9 | ROCm 5.0 |  |
| Hayabusa  | 2 * 16 * MI200 | PyTorch 1.9 | ROCm 5.0 |  |
| NC96_v4 | 1 * 4 * A100 PCIe | PyTorch 1.8 | CUDA 11.1 |  |

### Micro-benchmark Test
1. [x] Nccl-tests (NVIDIA only)
   > - [x]  GDR-only nccl-tests 
2. [x] Rccl-test (ROCm only)
   > - [x] update submodule to fix divide by zero error
3. [x] GPU-copy
  > - [x] Support bi-directional bandwidth benchmark
  > - [x] Support data checking and make it optional
4. [ ] GEMM benchmark (NVIDIA only)
  > - [ ] Support T4 and A10 in GEMM benchmark
5. [x]  GPU-burn benchmark (NVIDIA only)

### Model-benchmark Test
1. [x] Pytorch models
 > - [x]  Sync results on root rank for e2e model benchmarks in distributed mode 
 > - [x] Support customized `env` in local and torch.distributed mode 
 > - [x]  Add support for pytorch>=1.9.0 
 > - [x] Keep BatchNorm as fp32 for pytorch cnn models cast to fp16 
 > - [x]  Remove FP16 samples type converting time 
2. [x]  Support FAMBench 

###  Inference Benchmark Improvement
> - [x]  Add percentile metrics for ort and pytorch inference benchmarks 
> - [x]  Add configuration with inference benchmark

### SuperBench Improvement
> - [x] Add command to support listing all optional parameters for benchmarks. 
> - [x] Unify benchmark naming convention and support multiple tests with same benchmark and different parameters/options in one configuration file 
> - [x] Support timeout to detect the benchmark failure and stop the process automatically 
> - [x] Improve Output Interface

### Tools
1. [x] data diagnosis 
   > - [x] Support multi-benchmark check
   > - [x] Support output in excel and html format
   > - [x] Support result output for all nodes in data diagnosis
2. [x] Support result summary in excel ,md and  html format
